system-addon/lib/TopSitesFeed.jsm

```diff
@@ -25,6 +27,10 @@ this.TopSitesFeed = class TopSitesFeed {
     this.lastUpdated = 0;
     this._tippyTopProvider = new TippyTopProvider();
     this._tippyTopProvider.init();
+    this.dedupe = new Dedupe(this._dedupeKey);
+  }
+  _dedupeKey(site) {
+    return site.hostname;
```

Comment:
This is throwing an error and crashing the feed when the pinned group contains `null` items, so maybe check `site && site.hostname` and/or filter out `null` values from `pinned` before running dedupe